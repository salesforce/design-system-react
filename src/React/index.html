<!doctype html public "library">
<meta charset=utf-8/>
<head>
	<title>Selectlist</title>
	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
</head>
<body>
<script src="/dist/react/build.js"></script>
</body>
</html>