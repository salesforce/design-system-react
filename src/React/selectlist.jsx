//test

var React = require('react');
import {SelectlistCore} from "../SelectlistCore";
var DropdownButton = require('react-bootstrap/lib/DropdownButton');
var MenuItem = require('react-bootstrap/lib/MenuItem');

var DATA = {
	selected: 'None selected',
	id: 'selectlist1',
	items: [
		{ id: 0, name: 'tacos', type: 'mexican' },
		{ id: 1, name: 'burrito', type: 'mexican' },
		{ id: 2, name: 'tostada', type: 'mexican' },
		{ id: 3, name: 'hush puppies', type: 'southern' }
	]
};

var menuItemSelected = (eventKey, href, target)=> {
	DATA.selected = DATA.items[eventKey].name;
};
//Selectlist.handleMenuItemClicked
var menuItems = DATA.items.map( (menuItem) => {
	return <MenuItem eventKey={menuItem.id}>{menuItem.name}</MenuItem>;
});

var Selectlist = React.createClass({
	getInitialState: function() {
		return {
			selectedKey: undefined
		};
	},

	handleMenuItemClicked: function(eventKey, href, target) {
		this.setState({
			selectedKey: eventKey
		});
	},

	menuItems: function() {
		return DATA.items.map( (menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>{menuItem.name}</MenuItem>;
		});
	},

	selected: function() {
		return typeof(this.state.selectedKey) === 'undefined' ? 'None Selected' : DATA.items[this.state.selectedKey].name;
	},

	render () {
		return (
			<div>
				<h1>SelectList</h1>
				<ul>{this.menuItems}</ul>
				<DropdownButton title={this.selected()} key={DATA.id}>{this.menuItems()}</DropdownButton>
			</div>
		);
	}
});

React.render(<Selectlist/>, document.body, () => {});