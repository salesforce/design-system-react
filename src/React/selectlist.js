// SELECTLIST CONTROL

// core
import {Landmark} from '../Landmark';
import {SelectlistCore} from '../SelectlistCore';

// framework specific
import React from 'react';
import DropdownButton from 'react-bootstrap/lib/DropdownButton';
import MenuItem from 'react-bootstrap/lib/MenuItem';

export var Selectlist = React.createClass(Object.assign({}, SelectlistCore, {	
	propTypes: {
		disabled: React.PropTypes.bool,
		selection: React.PropTypes.number,
		collection: React.PropTypes.array
	},
	
	getInitialState () {
		return this.__getInitialState();
	},
		
	menuItems () {
		return this.props.collection.map((menuItem) => {
			return <MenuItem eventKey={menuItem.id} onSelect={this.handleMenuItemClicked}>
					{menuItem.name}
				</MenuItem>;
		});
	},
	
	componentWillReceiveProps(nextProps) {
		this.__initializeOptions(nextProps);
	},
	
	render () {
		var selection = this.getSelection();
		
		return (
			<DropdownButton className={this.cssClasses.CONTROL} disabled={this.state.disabled} title={selection ? selection.name : 'None selected'} key={this.props.id}>{this.menuItems()}</DropdownButton>
		);
	},
	
	componentWillMount () {
		this.elements = {
			wrapper: {
				toggleClass: function (cssClass, state) {
					Landmark.log(cssClass, state);
				}
			}
		}
		
		this.__constructor(this.props);
	},
	
	handleMenuItemClicked (eventKey, href, target) {
		this.setSelectionByKey('id', eventKey);
	}
}));