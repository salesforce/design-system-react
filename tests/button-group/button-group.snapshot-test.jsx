import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import { renderMarkup } from '../snapshot-helpers';

import IconGroup from '../../examples/button-group/icon-group';
import MoreIcon from '../../examples/button-group/more-icon';
import Checkbox from '../../examples/button-group/checkbox';
import CheckboxError from '../../examples/button-group/checkbox-error';

// Failing test, but unrelated to Checkbox Button Group which is the purpose of this pull request.

// test('Button Group IconGroup Snapshot', () => {
// 	const domTree = toJson(mount(
// 		<IconGroup />
// 	));
// 	expect(domTree).toMatchSnapshot();
// });

test('Button Group MoreIcon Snapshot', () => {
	const domTree = toJson(mount(
		<MoreIcon />
	));
	expect(domTree).toMatchSnapshot();
});

test('Button Group Checkbox Snapshot', () => {
	const domTree = toJson(mount(
		<Checkbox />
	));
	expect(domTree).toMatchSnapshot();
});

test('Button Group Checkbox Error Snapshot', () => {
	const domTree = toJson(mount(
		<CheckboxError />
	));
	expect(domTree).toMatchSnapshot();
});

test('Button Group Checkbox Error HTML Snapshot', () => {
	expect(renderMarkup(Checkbox)).toMatchSnapshot();
});

test('Button Group Checkbox Error HTML Snapshot', () => {
	expect(renderMarkup(CheckboxError)).toMatchSnapshot();
});
