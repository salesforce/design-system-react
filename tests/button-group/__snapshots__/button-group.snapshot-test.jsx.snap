exports[`test Button Group Checkbox Error HTML Snapshot 1`] = `
"<fieldset class=\"slds-form-element\">
  <legend class=\"slds-form-element__legend slds-form-element__label\">Scheduled Day(s)</legend>
  <div class=\"slds-form-element__control\">
    <div class=\"slds-checkbox--button-group\"><span class=\"slds-button slds-checkbox--button\"><input type=\"checkbox\" id=\"ButtonGroupExampleMon\"/><label class=\"slds-checkbox--button__label\" for=\"ButtonGroupExampleMon\"><span class=\"slds-checkbox--faux\">Mon</span></label>
      </span><span class=\"slds-button slds-checkbox--button\"><input type=\"checkbox\" id=\"ButtonGroupExampleTue\"/><label class=\"slds-checkbox--button__label\" for=\"ButtonGroupExampleTue\"><span class=\"slds-checkbox--faux\">Tue</span></label>
      </span><span class=\"slds-button slds-checkbox--button\"><input type=\"checkbox\" id=\"ButtonGroupExampleWed\"/><label class=\"slds-checkbox--button__label\" for=\"ButtonGroupExampleWed\"><span class=\"slds-checkbox--faux\">Wed</span></label>
      </span><span class=\"slds-button slds-checkbox--button\"><input type=\"checkbox\" id=\"ButtonGroupExampleThu\"/><label class=\"slds-checkbox--button__label\" for=\"ButtonGroupExampleThu\"><span class=\"slds-checkbox--faux\">Thu</span></label>
      </span><span class=\"slds-button slds-checkbox--button\"><input type=\"checkbox\" id=\"ButtonGroupExampleFri\"/><label class=\"slds-checkbox--button__label\" for=\"ButtonGroupExampleFri\"><span class=\"slds-checkbox--faux\">Fri</span></label>
      </span>
    </div>
  </div>
</fieldset>"
`;

exports[`test Button Group Checkbox Error HTML Snapshot 2`] = `
"<fieldset class=\"slds-form-element slds-has-error\">
  <legend class=\"slds-form-element__legend slds-form-element__label\">Scheduled Day(s)</legend>
  <div class=\"slds-form-element__control\">
    <div class=\"slds-checkbox--button-group\"><span class=\"slds-button slds-checkbox--button\"><input type=\"checkbox\" id=\"ButtonGroupExampleMon\"/><label class=\"slds-checkbox--button__label\" for=\"ButtonGroupExampleMon\"><span class=\"slds-checkbox--faux\">Mon</span></label>
      </span><span class=\"slds-button slds-checkbox--button\"><input type=\"checkbox\" id=\"ButtonGroupExampleTue\"/><label class=\"slds-checkbox--button__label\" for=\"ButtonGroupExampleTue\"><span class=\"slds-checkbox--faux\">Tue</span></label>
      </span><span class=\"slds-button slds-checkbox--button\"><input type=\"checkbox\" id=\"ButtonGroupExampleWed\"/><label class=\"slds-checkbox--button__label\" for=\"ButtonGroupExampleWed\"><span class=\"slds-checkbox--faux\">Wed</span></label>
      </span><span class=\"slds-button slds-checkbox--button\"><input type=\"checkbox\" id=\"ButtonGroupExampleThu\"/><label class=\"slds-checkbox--button__label\" for=\"ButtonGroupExampleThu\"><span class=\"slds-checkbox--faux\">Thu</span></label>
      </span><span class=\"slds-button slds-checkbox--button\"><input type=\"checkbox\" id=\"ButtonGroupExampleFri\"/><label class=\"slds-checkbox--button__label\" for=\"ButtonGroupExampleFri\"><span class=\"slds-checkbox--faux\">Fri</span></label>
      </span>
    </div>
    <div class=\"slds-form-element__help\">This field is required</div>
  </div>
</fieldset>"
`;

exports[`test Button Group Checkbox Error Snapshot 1`] = `
<ButtonGroupExample>
  <SLDSButtonGroup
    labels={
      Object {
        "error": "This field is required",
        "label": "Scheduled Day(s)",
      }
    }
    variant="checkbox">
    <fieldset
      className="slds-form-element slds-has-error">
      <legend
        className="slds-form-element__legend slds-form-element__label">
        Scheduled Day(s)
      </legend>
      <div
        className="slds-form-element__control">
        <div
          className="slds-checkbox--button-group">
          <SLDSCheckbox
            id="ButtonGroupExampleMon"
            label="Mon"
            labelToggleDisabled="Disabled"
            labelToggleEnabled="Enabled"
            variant="button-group">
            <span
              className="slds-button slds-checkbox--button">
              <input
                id="ButtonGroupExampleMon"
                onChange={[Function]}
                type="checkbox" />
              <label
                className="slds-checkbox--button__label"
                htmlFor="ButtonGroupExampleMon">
                <span
                  className="slds-checkbox--faux">
                  Mon
                </span>
              </label>
            </span>
          </SLDSCheckbox>
          <SLDSCheckbox
            id="ButtonGroupExampleTue"
            label="Tue"
            labelToggleDisabled="Disabled"
            labelToggleEnabled="Enabled"
            variant="button-group">
            <span
              className="slds-button slds-checkbox--button">
              <input
                id="ButtonGroupExampleTue"
                onChange={[Function]}
                type="checkbox" />
              <label
                className="slds-checkbox--button__label"
                htmlFor="ButtonGroupExampleTue">
                <span
                  className="slds-checkbox--faux">
                  Tue
                </span>
              </label>
            </span>
          </SLDSCheckbox>
          <SLDSCheckbox
            id="ButtonGroupExampleWed"
            label="Wed"
            labelToggleDisabled="Disabled"
            labelToggleEnabled="Enabled"
            variant="button-group">
            <span
              className="slds-button slds-checkbox--button">
              <input
                id="ButtonGroupExampleWed"
                onChange={[Function]}
                type="checkbox" />
              <label
                className="slds-checkbox--button__label"
                htmlFor="ButtonGroupExampleWed">
                <span
                  className="slds-checkbox--faux">
                  Wed
                </span>
              </label>
            </span>
          </SLDSCheckbox>
          <SLDSCheckbox
            id="ButtonGroupExampleThu"
            label="Thu"
            labelToggleDisabled="Disabled"
            labelToggleEnabled="Enabled"
            variant="button-group">
            <span
              className="slds-button slds-checkbox--button">
              <input
                id="ButtonGroupExampleThu"
                onChange={[Function]}
                type="checkbox" />
              <label
                className="slds-checkbox--button__label"
                htmlFor="ButtonGroupExampleThu">
                <span
                  className="slds-checkbox--faux">
                  Thu
                </span>
              </label>
            </span>
          </SLDSCheckbox>
          <SLDSCheckbox
            id="ButtonGroupExampleFri"
            label="Fri"
            labelToggleDisabled="Disabled"
            labelToggleEnabled="Enabled"
            variant="button-group">
            <span
              className="slds-button slds-checkbox--button">
              <input
                id="ButtonGroupExampleFri"
                onChange={[Function]}
                type="checkbox" />
              <label
                className="slds-checkbox--button__label"
                htmlFor="ButtonGroupExampleFri">
                <span
                  className="slds-checkbox--faux">
                  Fri
                </span>
              </label>
            </span>
          </SLDSCheckbox>
        </div>
        <div
          className="slds-form-element__help">
          This field is required
        </div>
      </div>
    </fieldset>
  </SLDSButtonGroup>
</ButtonGroupExample>
`;

exports[`test Button Group Checkbox Snapshot 1`] = `
<ButtonGroupExample>
  <SLDSButtonGroup
    labels={
      Object {
        "label": "Scheduled Day(s)",
      }
    }
    variant="checkbox">
    <fieldset
      className="slds-form-element">
      <legend
        className="slds-form-element__legend slds-form-element__label">
        Scheduled Day(s)
      </legend>
      <div
        className="slds-form-element__control">
        <div
          className="slds-checkbox--button-group">
          <SLDSCheckbox
            id="ButtonGroupExampleMon"
            label="Mon"
            labelToggleDisabled="Disabled"
            labelToggleEnabled="Enabled"
            variant="button-group">
            <span
              className="slds-button slds-checkbox--button">
              <input
                id="ButtonGroupExampleMon"
                onChange={[Function]}
                type="checkbox" />
              <label
                className="slds-checkbox--button__label"
                htmlFor="ButtonGroupExampleMon">
                <span
                  className="slds-checkbox--faux">
                  Mon
                </span>
              </label>
            </span>
          </SLDSCheckbox>
          <SLDSCheckbox
            id="ButtonGroupExampleTue"
            label="Tue"
            labelToggleDisabled="Disabled"
            labelToggleEnabled="Enabled"
            variant="button-group">
            <span
              className="slds-button slds-checkbox--button">
              <input
                id="ButtonGroupExampleTue"
                onChange={[Function]}
                type="checkbox" />
              <label
                className="slds-checkbox--button__label"
                htmlFor="ButtonGroupExampleTue">
                <span
                  className="slds-checkbox--faux">
                  Tue
                </span>
              </label>
            </span>
          </SLDSCheckbox>
          <SLDSCheckbox
            id="ButtonGroupExampleWed"
            label="Wed"
            labelToggleDisabled="Disabled"
            labelToggleEnabled="Enabled"
            variant="button-group">
            <span
              className="slds-button slds-checkbox--button">
              <input
                id="ButtonGroupExampleWed"
                onChange={[Function]}
                type="checkbox" />
              <label
                className="slds-checkbox--button__label"
                htmlFor="ButtonGroupExampleWed">
                <span
                  className="slds-checkbox--faux">
                  Wed
                </span>
              </label>
            </span>
          </SLDSCheckbox>
          <SLDSCheckbox
            id="ButtonGroupExampleThu"
            label="Thu"
            labelToggleDisabled="Disabled"
            labelToggleEnabled="Enabled"
            variant="button-group">
            <span
              className="slds-button slds-checkbox--button">
              <input
                id="ButtonGroupExampleThu"
                onChange={[Function]}
                type="checkbox" />
              <label
                className="slds-checkbox--button__label"
                htmlFor="ButtonGroupExampleThu">
                <span
                  className="slds-checkbox--faux">
                  Thu
                </span>
              </label>
            </span>
          </SLDSCheckbox>
          <SLDSCheckbox
            id="ButtonGroupExampleFri"
            label="Fri"
            labelToggleDisabled="Disabled"
            labelToggleEnabled="Enabled"
            variant="button-group">
            <span
              className="slds-button slds-checkbox--button">
              <input
                id="ButtonGroupExampleFri"
                onChange={[Function]}
                type="checkbox" />
              <label
                className="slds-checkbox--button__label"
                htmlFor="ButtonGroupExampleFri">
                <span
                  className="slds-checkbox--faux">
                  Fri
                </span>
              </label>
            </span>
          </SLDSCheckbox>
        </div>
      </div>
    </fieldset>
  </SLDSButtonGroup>
</ButtonGroupExample>
`;

exports[`test Button Group MoreIcon Snapshot 1`] = `
<ButtonGroupExample>
  <SLDSButtonGroup
    labels={Object {}}>
    <div
      className="slds-button-group"
      role="group">
      <SLDSButton
        disabled={false}
        hint={false}
        iconCategory="utility"
        iconSize="medium"
        label="Refresh"
        responsive={false}
        variant="neutral">
        <button
          className="slds-button slds-button--neutral"
          disabled={false}
          onClick={[Function]}>
          Refresh
        </button>
      </SLDSButton>
      <SLDSButton
        disabled={false}
        hint={false}
        iconCategory="utility"
        iconSize="medium"
        label="Edit"
        responsive={false}
        variant="neutral">
        <button
          className="slds-button slds-button--neutral"
          disabled={false}
          onClick={[Function]}>
          Edit
        </button>
      </SLDSButton>
      <SLDSButton
        disabled={false}
        hint={false}
        iconCategory="utility"
        iconSize="medium"
        label="Save"
        responsive={false}
        variant="neutral">
        <button
          className="slds-button slds-button--neutral"
          disabled={false}
          onClick={[Function]}>
          Save
        </button>
      </SLDSButton>
      <OnClickOutside(SLDSMenuDropdown)
        assistiveText="More Options"
        buttonVariant="icon"
        iconName="down"
        iconVariant="border-filled"
        id="ButtonGroupExampleDropdown"
        onSelect={[Function]}
        openOn="click"
        options={
          Array [
            Object {
              "label": "undo",
              "value": "A0",
            },
            Object {
              "label": "redo",
              "value": "B0",
            },
            Object {
              "label": "activate",
              "value": "C0",
            },
          ]
        }
        triggerClassName="slds-button--last">
        <SLDSMenuDropdown
          align="left"
          assistiveText="More Options"
          buttonVariant="icon"
          disableOnClickOutside={[Function]}
          enableOnClickOutside={[Function]}
          hoverCloseDelay={300}
          iconName="down"
          iconVariant="border-filled"
          id="ButtonGroupExampleDropdown"
          onSelect={[Function]}
          openOn="click"
          options={
            Array [
              Object {
                "label": "undo",
                "value": "A0",
              },
              Object {
                "label": "redo",
                "value": "B0",
              },
              Object {
                "label": "activate",
                "value": "C0",
              },
            ]
          }
          triggerClassName="slds-button--last">
          <SLDSMenuDropdownTrigger
            aria-haspopup={true}
            assistiveText="More Options"
            className="ignore-click-ButtonGroupExampleDropdown"
            iconName="down"
            iconVariant="border-filled"
            id="ButtonGroupExampleDropdown"
            isOpen={false}
            menu={null}
            onClick={[Function]}
            onFocus={null}
            onKeyDown={[Function]}
            onMouseEnter={null}
            onMouseLeave={null}
            openOn="click"
            tabIndex="0"
            triggerClassName="slds-button--last"
            triggerRef={[Function]}
            variant="icon">
            <div
              className="slds-dropdown-trigger slds-dropdown-trigger--click slds-button--last"
              id="ButtonGroupExampleDropdown"
              onClick={[Function]}
              onFocus={null}
              onKeyDown={[Function]}
              onMouseEnter={null}
              onMouseLeave={null}>
              <SLDSButton
                aria-expanded={false}
                aria-haspopup={true}
                assistiveText="More Options"
                className="ignore-click-ButtonGroupExampleDropdown"
                disabled={false}
                hint={false}
                iconCategory="utility"
                iconName="down"
                iconSize="medium"
                iconVariant="border-filled"
                responsive={false}
                tabIndex="0"
                variant="icon">
                <button
                  aria-expanded={false}
                  aria-haspopup={true}
                  className="slds-button slds-button--icon-border-filled ignore-click-ButtonGroupExampleDropdown"
                  disabled={false}
                  onClick={[Function]}
                  tabIndex="0">
                  <SLDSButtonIcon
                    category="utility"
                    className=""
                    hint={false}
                    name="down"
                    size={null}>
                    <UtilityIcon
                      aria-hidden="true"
                      category="utility"
                      className="slds-button__icon"
                      name="down">
                      <Svg
                        aria-hidden="true"
                        className="slds-button__icon"
                        data={
                          Object {
                            "path": Object {
                              "d": "M3.8 6.5h16.4c.4 0 .8.6.4 1l-8 9.8c-.3.3-.9.3-1.2 0l-8-9.8c-.4-.4-.1-1 .4-1z",
                            },
                            "viewBox": "0 0 24 24",
                          }
                        }
                        name="down">
                        <svg
                          aria-hidden="true"
                          className="slds-button__icon"
                          name="down"
                          viewBox="0 0 24 24">
                          <path
                            d="M3.8 6.5h16.4c.4 0 .8.6.4 1l-8 9.8c-.3.3-.9.3-1.2 0l-8-9.8c-.4-.4-.1-1 .4-1z" />
                        </svg>
                      </Svg>
                    </UtilityIcon>
                  </SLDSButtonIcon>
                  <span
                    className="slds-assistive-text">
                    More Options
                  </span>
                </button>
              </SLDSButton>
            </div>
          </SLDSMenuDropdownTrigger>
        </SLDSMenuDropdown>
      </OnClickOutside(SLDSMenuDropdown)>
    </div>
  </SLDSButtonGroup>
</ButtonGroupExample>
`;
